### DISCUSSION TITLE
StructuredArguments with Exceptions

### DISCUSSION THREAD

**Discussion by jeff-miller-cfa** [logfellow/logstash-logback-encoder#907] [Q&A] [answered]
What is the best way to log StructuredArguments and Exceptions? When you call `log.error("An error occurred", ex)` the exception is tracked in the event object but what if I wanted to add a StructuredArgument to that log? `log.error("An {} error occurred", value("errorType", errorType), ex)` will treat the exception as an argument not an exception.

**Comment by philsttr** [comment, marked as answer]
Log exceptions as you [normally would with slf4j](https://www.baeldung.com/slf4j-log-exceptions).  Slf4j/logback will interpret the last argument as the exception.  (The example you gave is correct)

**Reply by jeff-miller-cfa** [reply]
Awesome. Thanks for the response!
